File: summer/WebContent/js/system/windows/SingleChildEnumerator.js
/**
 * SingleChildEnumerator
 */

define(["dojo/_base/declare", "system/Type", "collections/IEnumerator"], function(declare, Type, IEnumerator){
	var SingleChildEnumerator = declare("SingleChildEnumerator", IEnumerator,{
		constructor:function(/*object*/ Child)
        {
            this._child = Child; 
            this._count = Child == null ? 0 : 1;
            this._index = -1;
        },
        
//        bool IEnumerator.
        MoveNext:function() 
        {
            this._index++; 
            return this._index < this._count; 
        },
 
//        void IEnumerator.
        Reset:function()
        {
            this._index = -1;
        } 
	});
	
	Object.defineProperties(SingleChildEnumerator.prototype,{
//        object IEnumerator.
        Current:
        { 
            get:function() { return (this._index == 0) ? this._child : null; }
        }
	});
	
	SingleChildEnumerator.Type = new Type("SingleChildEnumerator", SingleChildEnumerator, [IEnumerator.Type]);
	return SingleChildEnumerator;
});

File: org.summer.view.widget/src/org/summer/view/widget/SingleChildEnumerator.java
package org.summer.view.widget;
/*internal*/ public class SingleChildEnumerator implements IEnumerator 
{
    /*internal*/ public SingleChildEnumerator(Object Child)
    {
        _child = Child; 
        _count = Child == null ? 0 : 1;
    } 

    Object IEnumerator.Current
    { 
        get { return (_index == 0) ? _child : null; }
    }

    boolean IEnumerator.MoveNext() 
    {
        _index++; 
        return _index < _count; 
    }

    void IEnumerator.Reset()
    {
        _index = -1;
    } 

    private int _index = -1; 
    private int _count = 0; 
    private Object _child;
}